idea/vara-eth/frontend/src/features/codes/lib/queries.ts

```diff
@@ -1,56 +1,47 @@
 import { useQuery } from '@tanstack/react-query';
-import { request } from 'graphql-request';
 
 import { EXPLORER_URL } from '@/shared/config';
-import { graphql } from '@/shared/graphql';
+import { fetchWithGuard } from '@/shared/helpers';
+import { PaginatedResponse } from '@/shared/types';
 
 export const CODE_STATUS = {
-  VALIDATION_REQUESTED: 'validation_requested',
-  VALIDATION_FAILED: 'validation_failed',
-  VALIDATED: 'validated',
+  VALIDATION_REQUESTED: 'ValidationRequested',
+  VALIDATION_FAILED: 'ValidationFailed',
+  VALIDATED: 'Validated',
 } as const;
 
-const GET_CODES_QUERY = graphql(`
-  query GetCodes($first: Int!, $offset: Int!) {
-    allCodes(first: $first, offset: $offset) {
-      nodes {
-        id
-        status
-      }
-      totalCount
-    }
-  }
-`);
+export type Code = {
+  id: string;
+  status: string;
+  createdAt: string;
+};
+
+type CodesResponse = PaginatedResponse<Code>;
 
 export const useGetAllCodesQuery = (page: number, pageSize: number) => {
-  const first = pageSize;
+  const limit = pageSize;
   const offset = (page - 1) * pageSize;
 
   return useQuery({
-    queryKey: ['allCodes', first, offset],
-    queryFn: () => request(EXPLORER_URL, GET_CODES_QUERY, { first, offset }),
-    select: (data) => data.allCodes,
+    queryKey: ['allCodes', limit, offset],
+    queryFn: async () => {
+      const url = new URL(`${EXPLORER_URL}/codes`);
+      url.searchParams.set('limit', String(limit));
+      url.searchParams.set('offset', String(offset));
+
+      return fetchWithGuard<CodesResponse>({ url });
+    },
     placeholderData: (previousData) => previousData,
   });
 };
 
-const GET_CODE_BY_ID_QUERY = graphql(`
-  query GetCodeById($id: String!) {
-    codeById(id: $id) {
-      id
-      status
-    }
-  }
-`);
-
 export const useGetCodeByIdQuery = (id: string) => {
   return useQuery({
     queryKey: ['codeById', id],
     queryFn: async () => {
-      const result = await request(EXPLORER_URL, GET_CODE_BY_ID_QUERY, { id });
-      return result;
+      const url = new URL(`${EXPLORER_URL}/codes/${id}`);
```

Comment:
Any particular reason to use `new URL` instead of `string`? If there're currently no plans to have search params for urls like that, I believe better to keep things simple.